Automattic/wp-calypso
# Launchpad: Integrate Launchpad Checklist with Checklist API

### Important
- This is a trial PR for discussion.
- While this PR works, to work fully and correctly with launchpad, it requires simultaneous changes to wp-content/lib/onboarding-checklist to work. We need to be able to add/customize the checklist on the backend to be launchpad specific. This will also require site's site_intent option to be set to a launchpad flow ("newsletter", "link-in-bio", etc) to work (or some other backend option). 

### Proposed Changes
This PR integrates the existing Checklist API with Launchpad. When done, we can add, update, and manage checklist tasks via the same Checklist API that is used for the Home Site Checklist and other modules. Specifically, the following changes have been made: 
- Remove the tasks.tsx file that previously held hard-coded launchpad tasks.
- Add launchpad tasks to CHECKLIST_KNOWN_TASKS
- Add task transformations to the getTasks() method used by the home site checklist. This alters the properties of tasks so they can be consumed by home site checklist and now by launchpad. 
- Refactor the Launchpad Checklist module to pull in tasks via the useSiteChecklist hook
- Few other small adjustments to Checklist and Checklist Item components to accommodate standard task format (ie, we now use actionText instead of title for each list item. 

### Testing Instructions

This can only be partially tested. If you checkout this branch and test right now, the Launchpad will show whatever tasks your test site as part of its site checklist. Mine looks like this: 

<img width="1508" alt="Screenshot on 2022-08-19 at 09-45-21" src="https://user-images.githubusercontent.com/21228350/185693500-69255bb6-fe77-41d6-bc7d-9461bc8a009e.png">

To get this to show Launchpad specific tasks, we'll need to make the backend changes I mention above under Important. 

If you do want to test it as is, you can do the following: 
* Checkout this branch
* Run `yarn start`
* Go to http://calypso.localhost:3000/setup/launchpad?flow=newsletter&siteSlug=ericktesting5.wordpress.com (you can replace ericktesting5.wordpress.com with any WP simple site URL)

You should see Launchpad with a list of tasks. As noted, the tasks will be whatever is defined as the current site checklist. A separate PR will update backend logic to show Launchpad specific tasks in the right conditions.
